From 996d82b8533de70632623dc033d3990576d0f6aa Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Tue, 19 Jan 2016 16:45:25 +0100
Subject: [PATCH 069/201] Snap: Don't enable ZSL when disabling HDR

Instead, restore the default value. This prevents issues on devices
that don't support ZSL.

Change-Id: I9c456cd6503df29800e2bc7ccaa691cb3fb42681
---
 src/com/android/camera/PhotoModule.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 846c4319a..1e667b083 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -4829,6 +4829,7 @@ public class PhotoModule
         //filter off unsupported settings
         final String settingOff = mActivity.getString(R.string.setting_off_value);
         final String settingOn = mActivity.getString(R.string.setting_on_value);
+        final String zsl = mActivity.getString(R.string.pref_camera_zsl_default);
         if (!CameraSettings.isZSLHDRSupported(mParameters)) {
             //HDR internally uses AE-bracketing. Disable both if not supported.
             if (notSame(pref, CameraSettings.KEY_CAMERA_HDR, settingOff) ||
@@ -4839,7 +4840,7 @@ public class PhotoModule
                 mUI.setPreference(CameraSettings.KEY_AE_BRACKET_HDR, settingOff);
             } else if (notSame(pref, CameraSettings.KEY_CAMERA_HDR, settingOn) ||
                 notSame(pref, CameraSettings.KEY_AE_BRACKET_HDR, settingOn)) {
-                mUI.setPreference(CameraSettings.KEY_ZSL,settingOn);
+                mUI.setPreference(CameraSettings.KEY_ZSL, zsl);
             }
         }
 
-- 
2.17.1

